devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
Instead of polluting the code with this sort of 
``if self.profiler`` "do this" ``else`` "do something else"
I think there's a better way, but I'm interested in @mlange05 thought as well.

So here's what I propose.

We add a ``__new__`` method to ``StencilKernel``. If ``profile == True``, then we simply return a standard ``StencilKernel``, which as by default now, **does profile the C code**. Otherwise, we return a ``StencilKernelBasic`` -- please, find a better name -- which does "less" then a standard ``StencilKernel``. In particular, it does **not** perform C-level profiling. This is easily achievable by having an empty ``_profile_sections()`` and by tweaking the other functions accordingly. 

This should be useful for other future features as well (not just profiling), while leaving the implementation cleaner. 

@mloubout Of course, it's not just these three ``if self.profile`` that were worrying me, but rather the fact that if we don't deal with this **general** problem now (the fact that we may wanna have a ``StencilKernel`` for normal execution and a ``StencilKernelBasic`` for debug/... mode), then the situation might escalate badly rapidly